independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/ClientRequestContextImpl.java

```diff
@@ -63,16 +62,10 @@ public ClientRequestContextImpl(RestClientRequestContext restClientRequestContex
         this.headersMap = new ClientRequestHeadersMap(); //restClientRequestContext.requestHeaders.getHeaders()
         this.providers = new ProvidersImpl(restClientRequestContext);
 
-        // TODO This needs to be challenged:
         // Always create a duplicated context because each REST Client invocation must have its own context
         // A separate context allows integrations like OTel to create a separate Span for each invocation (expected)
-        Context ctxt = Vertx.currentContext();
-        if (ctxt != null && VertxContext.isDuplicatedContext(ctxt)) {
-            this.context = ctxt;
-        } else {
-            Context current = client.vertx.getOrCreateContext();
-            this.context = VertxContext.createNewDuplicatedContext(current);
-        }
+        Context current = client.vertx.getOrCreateContext();
+        this.context = VertxContext.createNewDuplicatedContext(current);
```

Comment:
I'm actually unsure of this. I would really like to know if the tests pass without this change

Reply:
It fails, I tried the test with no main changes.

Reply:
Note that if I do not ensure that a DuplicateContext instance is created, the result in the trace is still inconsistent.

Now, without modifying this logic and only keeping the changes made in HandlerChain, the only thing different is that there are two traces where `"operationName": "GET /step_3"`. but still inconsistent.

<details><summary>Trace Data</summary>
<p>

```
{
  "data": [
    {
      "traceID": "f839289a74ed082f3839bc6968765708",
      "spans": [
        {
          "traceID": "f839289a74ed082f3839bc6968765708",
          "spanID": "f1cf382a3d83054f",
          "operationName": "GET /wizard-server/start",
          "references": [
            {
              "refType": "CHILD_OF",
              "traceID": "f839289a74ed082f3839bc6968765708",
              "spanID": "298e03112df3e297"
            }
          ],
          "startTime": 1731672920394667,
          "duration": 2020218,
          "tags": [
            {
              "key": "server.address",
              "type": "string",
              "value": "localhost"
            },
            {
              "key": "client.address",
              "type": "string",
              "value": "127.0.0.1"
            },
            {
              "key": "user_agent.original",
              "type": "string",
              "value": "Quarkus REST Client"
            },
            {
              "key": "url.path",
              "type": "string",
              "value": "/wizard-server/start"
            },
            {
              "key": "code.namespace",
              "type": "string",
              "value": "org.acme.server.WizardServerResource"
            },
            {
              "key": "http.response.body.size",
              "type": "int64",
              "value": 10
            },
            {
              "key": "http.request.method",
              "type": "string",
              "value": "GET"
            },
            {
              "key": "code.function",
              "type": "string",
              "value": "start"
            },
            {
              "key": "http.route",
              "type": "string",
              "value": "/wizard-server/start"
            },
            {
              "key": "server.port",
              "type": "int64",
              "value": 8080
            },
            {
              "key": "url.scheme",
              "type": "string",
              "value": "http"
            },
            {
              "key": "http.response.status_code",
              "type": "int64",
              "value": 200
            },
            {
              "key": "span.kind",
              "type": "string",
              "value": "server"
            },
            {
              "key": "internal.span.format",
              "type": "string",
              "value": "otlp"
            }
          ],
          "logs": [],
          "processID": "p1",
          "warnings": null
        },
        {
          "traceID": "f839289a74ed082f3839bc6968765708",
          "spanID": "9aabd1c958cde2cc",
          "operationName": "GET /step_3",
          "references": [
            {
              "refType": "CHILD_OF",
              "traceID": "f839289a74ed082f3839bc6968765708",
              "spanID": "97a9eae6612ae05c"
            }
          ],
          "startTime": 1731672922433087,
          "duration": 1004952,
          "tags": [
            {
              "key": "server.address",
              "type": "string",
              "value": "localhost"
            },
            {
              "key": "http.request.method",
              "type": "string",
              "value": "GET"
            },
            {
              "key": "server.port",
              "type": "int64",
              "value": 8080
            },
            {
              "key": "url.full",
              "type": "string",
              "value": "http://localhost:8080/wizard-server/step_3"
            },
            {
              "key": "http.response.status_code",
              "type": "int64",
              "value": 200
            },
            {
              "key": "span.kind",
              "type": "string",
              "value": "client"
            },
            {
              "key": "internal.span.format",
              "type": "string",
              "value": "otlp"
            }
          ],
          "logs": [],
          "processID": "p1",
          "warnings": null
        },
        {
          "traceID": "f839289a74ed082f3839bc6968765708",
          "spanID": "298e03112df3e297",
          "operationName": "GET /start",
          "references": [
            {
              "refType": "CHILD_OF",
              "traceID": "f839289a74ed082f3839bc6968765708",
              "spanID": "97a9eae6612ae05c"
            }
          ],
          "startTime": 1731672920392373,
          "duration": 2026956,
          "tags": [
            {
              "key": "server.address",
              "type": "string",
              "value": "localhost"
            },
            {
              "key": "http.request.method",
              "type": "string",
              "value": "GET"
            },
            {
              "key": "server.port",
              "type": "int64",
              "value": 8080
            },
            {
              "key": "url.full",
              "type": "string",
              "value": "http://localhost:8080/wizard-server/start"
            },
            {
              "key": "http.response.status_code",
              "type": "int64",
              "value": 200
            },
            {
              "key": "span.kind",
              "type": "string",
              "value": "client"
            },
            {
              "key": "internal.span.format",
              "type": "string",
              "value": "otlp"
            }
          ],
          "logs": [],
          "processID": "p1",
          "warnings": null
        },
        {
          "traceID": "f839289a74ed082f3839bc6968765708",
          "spanID": "34bbf1dbe559e321",
          "operationName": "GET /wizard-server/step_3",
          "references": [
            {
              "refType": "CHILD_OF",
              "traceID": "f839289a74ed082f3839bc6968765708",
              "spanID": "9aabd1c958cde2cc"
            }
          ],
          "startTime": 1731672922434013,
          "duration": 1003499,
          "tags": [
            {
              "key": "server.address",
              "type": "string",
              "value": "localhost"
            },
            {
              "key": "client.address",
              "type": "string",
              "value": "127.0.0.1"
            },
            {
              "key": "user_agent.original",
              "type": "string",
              "value": "Quarkus REST Client"
            },
            {
              "key": "url.path",
              "type": "string",
              "value": "/wizard-server/step_3"
            },
            {
              "key": "code.namespace",
              "type": "string",
              "value": "org.acme.server.WizardServerResource"
            },
            {
              "key": "http.response.body.size",
              "type": "int64",
              "value": 11
            },
            {
              "key": "http.request.method",
              "type": "string",
              "value": "GET"
            },
            {
              "key": "code.function",
              "type": "string",
              "value": "step3"
            },
            {
              "key": "http.route",
              "type": "string",
              "value": "/wizard-server/step_3"
            },
            {
              "key": "server.port",
              "type": "int64",
              "value": 8080
            },
            {
              "key": "url.scheme",
              "type": "string",
              "value": "http"
            },
            {
              "key": "http.response.status_code",
              "type": "int64",
              "value": 200
            },
            {
              "key": "span.kind",
              "type": "string",
              "value": "server"
            },
            {
              "key": "internal.span.format",
              "type": "string",
              "value": "otlp"
            }
          ],
          "logs": [],
          "processID": "p1",
          "warnings": null
        },
        {
          "traceID": "f839289a74ed082f3839bc6968765708",
          "spanID": "f4af771e76197e50",
          "operationName": "GET /wizard-server/step_1",
          "references": [
            {
              "refType": "CHILD_OF",
              "traceID": "f839289a74ed082f3839bc6968765708",
              "spanID": "e83972f0306f2118"
            }
          ],
          "startTime": 1731672922432499,
          "duration": 1007849,
          "tags": [
            {
              "key": "server.address",
              "type": "string",
              "value": "localhost"
            },
            {
              "key": "client.address",
              "type": "string",
              "value": "127.0.0.1"
            },
            {
              "key": "user_agent.original",
              "type": "string",
              "value": "Quarkus REST Client"
            },
            {
              "key": "url.path",
              "type": "string",
              "value": "/wizard-server/step_1"
            },
            {
              "key": "code.namespace",
              "type": "string",
              "value": "org.acme.server.WizardServerResource"
            },
            {
              "key": "http.response.body.size",
              "type": "int64",
              "value": 11
            },
            {
              "key": "http.request.method",
              "type": "string",
              "value": "GET"
            },
            {
              "key": "code.function",
              "type": "string",
              "value": "step1"
            },
            {
              "key": "http.route",
              "type": "string",
              "value": "/wizard-server/step_1"
            },
            {
              "key": "server.port",
              "type": "int64",
              "value": 8080
            },
            {
              "key": "url.scheme",
              "type": "string",
              "value": "http"
            },
            {
              "key": "http.response.status_code",
              "type": "int64",
              "value": 200
            },
            {
              "key": "span.kind",
              "type": "string",
              "value": "server"
            },
            {
              "key": "internal.span.format",
              "type": "string",
              "value": "otlp"
            }
          ],
          "logs": [],
          "processID": "p1",
          "warnings": null
        },
        {
          "traceID": "f839289a74ed082f3839bc6968765708",
          "spanID": "e83972f0306f2118",
          "operationName": "GET /step_1",
          "references": [
            {
              "refType": "CHILD_OF",
              "traceID": "f839289a74ed082f3839bc6968765708",
              "spanID": "97a9eae6612ae05c"
            }
          ],
          "startTime": 1731672922431589,
          "duration": 1009121,
          "tags": [
            {
              "key": "server.address",
              "type": "string",
              "value": "localhost"
            },
            {
              "key": "http.request.method",
              "type": "string",
              "value": "GET"
            },
            {
              "key": "server.port",
              "type": "int64",
              "value": 8080
            },
            {
              "key": "url.full",
              "type": "string",
              "value": "http://localhost:8080/wizard-server/step_1"
            },
            {
              "key": "http.response.status_code",
              "type": "int64",
              "value": 200
            },
            {
              "key": "span.kind",
              "type": "string",
              "value": "client"
            },
            {
              "key": "internal.span.format",
              "type": "string",
              "value": "otlp"
            }
          ],
          "logs": [],
          "processID": "p1",
          "warnings": null
        },
        {
          "traceID": "f839289a74ed082f3839bc6968765708",
          "spanID": "97a9eae6612ae05c",
          "operationName": "GET /wizard-client/work",
          "references": [],
          "startTime": 1731672920336452,
          "duration": 7116351,
          "tags": [
            {
              "key": "server.address",
              "type": "string",
              "value": "localhost"
            },
            {
              "key": "client.address",
              "type": "string",
              "value": "127.0.0.1"
            },
            {
              "key": "url.path",
              "type": "string",
              "value": "/wizard-client/work"
            },
            {
              "key": "code.namespace",
              "type": "string",
              "value": "org.acme.client.WizardClientResource"
            },
            {
              "key": "http.response.body.size",
              "type": "int64",
              "value": 10
            },
            {
              "key": "http.request.method",
              "type": "string",
              "value": "GET"
            },
            {
              "key": "http.route",
              "type": "string",
              "value": "/wizard-client/work"
            },
            {
              "key": "user_agent.original",
              "type": "string",
              "value": "IntelliJ HTTP Client/IntelliJ IDEA 2024.2.3"
            },
            {
              "key": "code.function",
              "type": "string",
              "value": "work"
            },
            {
              "key": "server.port",
              "type": "int64",
              "value": 8080
            },
            {
              "key": "http.request.body.size",
              "type": "int64",
              "value": 0
            },
            {
              "key": "url.scheme",
              "type": "string",
              "value": "http"
            },
            {
              "key": "http.response.status_code",
              "type": "int64",
              "value": 200
            },
            {
              "key": "span.kind",
              "type": "string",
              "value": "server"
            },
            {
              "key": "internal.span.format",
              "type": "string",
              "value": "otlp"
            }
          ],
          "logs": [],
          "processID": "p1",
          "warnings": null
        },
        {
          "traceID": "f839289a74ed082f3839bc6968765708",
          "spanID": "ed26d8a51091431c",
          "operationName": "GET /wizard-server/step_2",
          "references": [
            {
              "refType": "CHILD_OF",
              "traceID": "f839289a74ed082f3839bc6968765708",
              "spanID": "6ef91de78192d16b"
            }
          ],
          "startTime": 1731672922433616,
          "duration": 4006163,
          "tags": [
            {
              "key": "server.address",
              "type": "string",
              "value": "localhost"
            },
            {
              "key": "client.address",
              "type": "string",
              "value": "127.0.0.1"
            },
            {
              "key": "user_agent.original",
              "type": "string",
              "value": "Quarkus REST Client"
            },
            {
              "key": "url.path",
              "type": "string",
              "value": "/wizard-server/step_2"
            },
            {
              "key": "code.namespace",
              "type": "string",
              "value": "org.acme.server.WizardServerResource"
            },
            {
              "key": "http.response.body.size",
              "type": "int64",
              "value": 11
            },
            {
              "key": "http.request.method",
              "type": "string",
              "value": "GET"
            },
            {
              "key": "code.function",
              "type": "string",
              "value": "step2"
            },
            {
              "key": "http.route",
              "type": "string",
              "value": "/wizard-server/step_2"
            },
            {
              "key": "server.port",
              "type": "int64",
              "value": 8080
            },
            {
              "key": "url.scheme",
              "type": "string",
              "value": "http"
            },
            {
              "key": "http.response.status_code",
              "type": "int64",
              "value": 200
            },
            {
              "key": "span.kind",
              "type": "string",
              "value": "server"
            },
            {
              "key": "internal.span.format",
              "type": "string",
              "value": "otlp"
            }
          ],
          "logs": [],
          "processID": "p1",
          "warnings": null
        },
        {
          "traceID": "f839289a74ed082f3839bc6968765708",
          "spanID": "b320954bb3eadbb7",
          "operationName": "GET /complete",
          "references": [
            {
              "refType": "CHILD_OF",
              "traceID": "f839289a74ed082f3839bc6968765708",
              "spanID": "97a9eae6612ae05c"
            }
          ],
          "startTime": 1731672926443614,
          "duration": 1007537,
          "tags": [
            {
              "key": "server.address",
              "type": "string",
              "value": "localhost"
            },
            {
              "key": "http.request.method",
              "type": "string",
              "value": "GET"
            },
            {
              "key": "server.port",
              "type": "int64",
              "value": 8080
            },
            {
              "key": "url.full",
              "type": "string",
              "value": "http://localhost:8080/wizard-server/complete"
            },
            {
              "key": "http.response.status_code",
              "type": "int64",
              "value": 200
            },
            {
              "key": "span.kind",
              "type": "string",
              "value": "client"
            },
            {
              "key": "internal.span.format",
              "type": "string",
              "value": "otlp"
            }
          ],
          "logs": [],
          "processID": "p1",
          "warnings": null
        },
        {
          "traceID": "f839289a74ed082f3839bc6968765708",
          "spanID": "a92ac52b7309b7e6",
          "operationName": "GET /wizard-server/complete",
          "references": [
            {
              "refType": "CHILD_OF",
              "traceID": "f839289a74ed082f3839bc6968765708",
              "spanID": "b320954bb3eadbb7"
            }
          ],
          "startTime": 1731672926444862,
          "duration": 1005187,
          "tags": [
            {
              "key": "server.address",
              "type": "string",
              "value": "localhost"
            },
            {
              "key": "client.address",
              "type": "string",
              "value": "127.0.0.1"
            },
            {
              "key": "user_agent.original",
              "type": "string",
              "value": "Quarkus REST Client"
            },
            {
              "key": "url.path",
              "type": "string",
              "value": "/wizard-server/complete"
            },
            {
              "key": "code.namespace",
              "type": "string",
              "value": "org.acme.server.WizardServerResource"
            },
            {
              "key": "http.response.body.size",
              "type": "int64",
              "value": 13
            },
            {
              "key": "http.request.method",
              "type": "string",
              "value": "GET"
            },
            {
              "key": "code.function",
              "type": "string",
              "value": "complete"
            },
            {
              "key": "http.route",
              "type": "string",
              "value": "/wizard-server/complete"
            },
            {
              "key": "server.port",
              "type": "int64",
              "value": 8080
            },
            {
              "key": "url.scheme",
              "type": "string",
              "value": "http"
            },
            {
              "key": "http.response.status_code",
              "type": "int64",
              "value": 200
            },
            {
              "key": "span.kind",
              "type": "string",
              "value": "server"
            },
            {
              "key": "internal.span.format",
              "type": "string",
              "value": "otlp"
            }
          ],
          "logs": [],
          "processID": "p1",
          "warnings": null
        },
        {
          "traceID": "f839289a74ed082f3839bc6968765708",
          "spanID": "6ef91de78192d16b",
          "operationName": "GET /step_3",
          "references": [
            {
              "refType": "CHILD_OF",
              "traceID": "f839289a74ed082f3839bc6968765708",
              "spanID": "97a9eae6612ae05c"
            }
          ],
          "startTime": 1731672922432678,
          "duration": 4007676,
          "tags": [
            {
              "key": "server.address",
              "type": "string",
              "value": "localhost"
            },
            {
              "key": "http.request.method",
              "type": "string",
              "value": "GET"
            },
            {
              "key": "server.port",
              "type": "int64",
              "value": 8080
            },
            {
              "key": "url.full",
              "type": "string",
              "value": "http://localhost:8080/wizard-server/step_2"
            },
            {
              "key": "http.response.status_code",
              "type": "int64",
              "value": 200
            },
            {
              "key": "span.kind",
              "type": "string",
              "value": "client"
            },
            {
              "key": "internal.span.format",
              "type": "string",
              "value": "otlp"
            }
          ],
          "logs": [],
          "processID": "p1",
          "warnings": null
        }
      ],
      "processes": {
        "p1": {
          "serviceName": "quarkus-open-telemetry-bug",
          "tags": [
            {
              "key": "host.name",
              "type": "string",
              "value": "....local"
            },
            {
              "key": "otel.library.name",
              "type": "string",
              "value": "io.quarkus.opentelemetry"
            },
            {
              "key": "service.version",
              "type": "string",
              "value": "1.0.0-SNAPSHOT"
            },
            {
              "key": "telemetry.sdk.language",
              "type": "string",
              "value": "java"
            },
            {
              "key": "telemetry.sdk.name",
              "type": "string",
              "value": "opentelemetry"
            },
            {
              "key": "telemetry.sdk.version",
              "type": "string",
              "value": "1.42.1"
            },
            {
              "key": "webengine.name",
              "type": "string",
              "value": "Quarkus"
            },
            {
              "key": "webengine.version",
              "type": "string",
              "value": "999-SNAPSHOT"
            }
          ]
        }
      },
      "warnings": null
    }
  ],
  "total": 0,
  "limit": 0,
  "offset": 0,
  "errors": null
}
``` 

</p>
</details>

Reply:
When ClientObservabilityHandler calls this constructor, It does this for each of the identified template paths. So if you reuse the instance, some ClientObservabilityHandler will override the ClientUrlPathTemplate which is used to construct the correct operationName value. 

https://github.com/quarkusio/quarkus/blob/main/independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientObservabilityHandler.java#L21

Reply:
In any case, you've done excellent work here! 

Initially we did do it way you are proposing here, but it was changed in #27629

Reply:
Sounds good. Thank you so much for your support. I appreciate it. 

I found that change (https://github.com/quarkusio/quarkus/commit/ac1a11e274b281c291933c394258d8caeae2ecc1) two days ago but I didn't find the reasons why it was reverted. Please let me know if you find a better solution.

Reply:
I will keep you updated